app/grandchallenge/algorithms/views.py

```diff
@@ -557,21 +560,50 @@ def create_upload_session(image_files):
                 civ = ci.create_instance(value=value)
                 component_interface_values.append(civ)
 
-        job = Job.objects.create(
-            creator=self.request.user,
-            algorithm_image=form.cleaned_data["algorithm_image"],
-            extra_logs_viewer_groups=[self.algorithm.editors_group],
-            input_civ_set=component_interface_values,
-            time_limit=self.algorithm.time_limit,
-        )
-        job.sort_inputs_and_execute(upload_session_pks=upload_session_pks)
+        # check that this job hasn't been run yet:
+        unique_kwargs = {
+            "algorithm_image": form.cleaned_data["algorithm_image"],
+        }
+        input_interface_count = self.algorithm.inputs.count()
+        if form.cleaned_data["algorithm_model"]:
+            unique_kwargs["algorithm_model"] = form.cleaned_data[
+                "algorithm_model"
+            ]
+
+        if (
+            Job.objects.filter(**unique_kwargs)
+            .annotate(
+                inputs_match_count=Count(
+                    "inputs",
+                    filter=Q(inputs__in=component_interface_values),
+                )
+            )
+            .filter(inputs_match_count=input_interface_count)
+            .exists()
+        ):
+            form.add_error(
+                None,
+                "A result for these inputs with the current image "
+                "and model already exists.",
+            )
+            return self.form_invalid(form)
```

Comment:
This is not the way to do it, but it requires a more extensive refactoring of CIV creation and validation for jobs to fix. That should be a seperate PR.

Reply:
Does the same need to be done for the job creation API view?

Reply:
Yes, good catch.

Reply:
Actually, it turns out that on the job serializer we always create new civs, so this check will always pass. Hence, not needed. We might want to update the serializer to not always create new civs, but that's for another PR.

Reply:
Could you create a separate issue for this?

Reply:
Done!